src/factories/calorimetry/CalorimeterClusterRecoCoG_factoryT.h

```diff
@@ -48,15 +59,25 @@ class CalorimeterClusterRecoCoG_factoryT :
         // SpdlogMixin logger initialization, sets m_log
         InitLogger(app, GetPrefix(), "info");
 
-        // Algorithm configuration
-        auto cfg = GetDefaultConfig();
-
-        app->SetDefaultParameter(param_prefix + ":samplingFraction", cfg.sampFrac);
-        app->SetDefaultParameter(param_prefix + ":logWeightBase", cfg.logWeightBase);
-        app->SetDefaultParameter(param_prefix + ":energyWeight", cfg.energyWeight);
-        app->SetDefaultParameter(param_prefix + ":enableEtaBounds", cfg.enableEtaBounds);
+        // Initialize properties
+        for (const auto& [key, prop] : m_algo.getProperties()) {
+          std::visit(
+            [app, &m_algo = m_algo, &m_log = m_log,
```

Comment:
It feels like we should be able to avoid the capture of m_algo here if we pass prop instead of prop.get().

Reply:
This isn't possible since visit needs the second argument to be the variant, not something that contains a variant...